packages/xo-server/src/xo-mixins/authentication.mjs

```diff
@@ -210,14 +209,7 @@ export default class {
   }
 
   async deleteAuthenticationTokens({ filter }) {
-    let predicate
-    const { apiContext } = this._app
-    if (apiContext !== undefined && apiContext.permission !== 'admin') {
-      predicate = { user_id: apiContext.user.id }
-    }
-
-    const db = this._tokens
-    return db.remove((await db.get(predicate)).filter(createPredicate(filter)).map(({ id }) => id))
+    await this._tokens.remove(filter)
```

Comment:
Can `.remove()` take a filter?

Reply:
Yes :slightly_smiling_face:

Reply:
It doesn't seem to work when I test it though :thinking: Where does it interpret the filter?
https://github.com/vatesfr/xen-orchestra/blob/f034ec45f36d9357304aa7bf203ef606bf57c068/packages/xo-collection/src/collection.js#L130-L135

Reply:
https://github.com/vatesfr/xen-orchestra/blob/f034ec45f36d9357304aa7bf203ef606bf57c068/packages/xo-server/src/collection.mjs#L49-L58

Reply:
I don't see where the filter gets interpreted there. :arrow_up: 

If I'm following the path correctly, it's interpreted there, but as a lodash filter, not as a Complex Matcher filter:
https://github.com/vatesfr/xen-orchestra/blob/1046bbdd99d4fcc5bf58fdbf8728f5806d2821c6/packages/xo-server/src/collection/redis.mjs#L195-L201

Also, does it work for you when you try the "Forget all connection tokens" feature? It's supposed to delete all the user's tokens except one, right? It doesn't seem to work on my side.

Reply:
My bad, you're right, it's not a value matcher filter, it's a basic property matching *à la* lodash, I'll fix it!